osu.Server.Queues.ScoreStatisticsProcessor/Helpers/WebRequestHelper.cs

```diff
@@ -0,0 +1,222 @@
+        /// If this ever becomes a problem the proposition is to move the ES lookup logic to https://github.com/ppy/osu-global-rank-lookup-cache and have both web and this call into that component.
+        /// </para>
+        /// <para>
+        /// Note that this working correctly also relies on the fact that <c>osu-web</c> API has lazer mode turned forcefully on with no way to toggle it off
+        /// (see https://github.com/ppy/osu-web/blob/f46806bb81eb0d3b0807c70bf8e3dc13f0783ec9/app/Libraries/Search/ScoreSearchParams.php#L86-L88).
+        /// </para>
+        /// </remarks>
+        public static int? GetScoreRankOnBeatmapLeaderboard(SoloScore score)
+        {
+            try
+            {
+                string? token = retrieveAccessToken();
+                if (token == null)
+                    return null;
+
+                var requestMsg = new HttpRequestMessage(HttpMethod.Get, $"{shared_interop_domain}/api/v2/scores/{score.id}");
+                requestMsg.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token);
+                var responseMsg = http.Send(requestMsg);
+
+                if (!responseMsg.IsSuccessStatusCode || responseMsg.Content.ReadFromJsonAsync<ScoreResponse>().Result is not ScoreResponse scoreResponse)
+                {
+                    SentrySdk.CaptureMessage($"Failed to retrieve score rank from API: received status code {responseMsg.StatusCode}, content: {responseMsg.Content.ReadAsStringAsync().Result}",
+                        SentryLevel.Warning);
+                    return null;
+                }
+
+                return scoreResponse.rank_global;
+            }
+            catch (Exception e)
+            {
+                SentrySdk.CaptureException(e, scope => scope.Level = SentryLevel.Warning);
+                return null;
+            }
+        }
+
+        private static string? retrieveAccessToken()
+        {
+            if (string.IsNullOrEmpty(api_client_id) || string.IsNullOrEmpty(api_client_secret))
+            {
+#if !DEBUG
+                throw new InvalidOperationException($"Attempted API call without client id and secret set!");
+#endif
+                return null;
+            }
+
+            if (accessToken != null && accessTokenExpiry != null && DateTimeOffset.Now <= accessTokenExpiry.Value.AddMinutes(-5))
+                return accessToken;
+
+            var requestMsg = new HttpRequestMessage(HttpMethod.Post, $"{shared_interop_domain}/oauth/token");
+            requestMsg.Content = new FormUrlEncodedContent([
+                new KeyValuePair<string, string>("client_id", api_client_id),
+                new KeyValuePair<string, string>("client_secret", api_client_secret),
+                new KeyValuePair<string, string>("grant_type", "client_credentials"),
+                new KeyValuePair<string, string>("scope", "public"),
+            ]);

```

Comment:
This sets up the authentication with the API as stated in the OP.

The lazer API client id/secret pair cannot be reused here for two reasons: that token only supports `*` scope (`public` scope [requires the client to have an user owning it](https://github.com/ppy/osu-web/blob/904c94936f2e4b87c88c2dbfd37e74b2b107fca1/app/Models/OAuth/Token.php#L183-L186)), and [the `*` scope doesn't support `client_credentials` grant](https://github.com/ppy/osu-web/blob/904c94936f2e4b87c88c2dbfd37e74b2b107fca1/app/Models/OAuth/Token.php#L189-L191).

So this needs to either be a user-owned client (by whom?) or I'm going to need @ppy/team-web to carve me out something here.